pkg/containertools/dockerfilegenerator.go

```diff
@@ -0,0 +1,98 @@
+package containertools
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/sirupsen/logrus"
+)
+
+const (
+	baseImage = "scratch"
+	defaultBinarySourceImage = "quay.io/kevinrizza/upstream-opm-builder"
+	defaultDbLocation = "./index.db"
+	dbLocationLabel = "operators.operatorframework.io.index.database.v1"
+)
+
+// DockerfileGenerator defines functions to generate index dockerfiles
+type DockerfileGenerator interface {
+	GenerateIndexDockerfile(string,string,[]string, []string) (string, error)
+}
+
+// DockerfileGeneratorImpl struct implementation of DockerfileGenerator interface
+type DockerfileGeneratorImpl struct {
+	logger *logrus.Entry
+	labelReader LabelReader
+}
+
+// NewDockerfileGenerator is a constructor that returns a DockerfileGenerator
+func NewDockerfileGenerator(containerTool string, logger *logrus.Entry) DockerfileGenerator {
+	return &DockerfileGeneratorImpl{
+		logger: logger,
+		labelReader: NewLabelReader(containerTool, logger),
+	}
+}
+
+// GenerateIndexDockerfile builds a string representation of a dockerfile to use when building
+// an operator-registry index image
+func (g *DockerfileGeneratorImpl) GenerateIndexDockerfile(binarySourceImage, previousIndexImage string, bundlesToAdd, operatorsToDelete []string) (string, error) {
```

Comment:
super-nit: my brain keeps reading "binary source image" as "source code written in binary" and not "the source image containing the binaries"

Reply:
Yeah I had a lot of trouble coming up with these names (there are actually 3 different images involved). I was hoping someone could tell me this is bad and have a better idea. Any suggestions?